File: frontend/src/types/expense.ts
export type Expense = {
    id: string;
    tripId: string;
    name: string;
    plannedCostId: string | null;
    category: string | null;
    amount: number;
    expenseDate: string | null;
    createdAtUtc: string;
};

export type NewExpenseForm = {
    name: string;
    category: string;
    amount: string;
    expenseDate: string;
    plannedCostId: string | null;
};

export const initialNewExpenseForm: NewExpenseForm = {
    name: "",
    category: "",
    amount: "",
    expenseDate: "",
    plannedCostId: null,
};

File: backend/Models/Expense.cs
namespace TravelAssistant.Models;

/// <summary>
/// Represents one actual expense recorded against a trip, optionally linked to the plan it fulfils.
/// </summary>
public class Expense
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid TripId { get; set; }

    public Trip Trip { get; set; } = null!;

    public string Name { get; set; } = "Cost item";

    public Guid? PlannedCostId { get; set; }

    public PlannedCost? PlannedCost { get; set; }

    public ExpenseCategory? Category { get; set; }

    public decimal Amount { get; set; }

    public DateOnly? ExpenseDate { get; set; }

    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
}
